derive/src/lib.rs

```diff
@@ -153,16 +153,24 @@ fn generate_fields(fields: &FieldsList) -> Vec<TokenStream2> {
 
             let type_name = clean_type_string(&quote!(#ty).to_string());
             let docs = utils::get_doc_literals(&f.attrs);
-            let method_call = if utils::is_compact(f) {
-                quote!(.compact_of::<#ty>)
+            let type_of_method = if utils::is_compact(f) {
+                quote!(compact)
             } else {
-                quote!(.field_of::<#ty>)
+                quote!(ty)
             };
-            if let Some(ident) = ident {
-                quote!(#method_call(stringify!(#ident), #type_name, &[ #( #docs ),* ]))
+            let name = if let Some(ident) = ident {
+                quote!(.name(stringify!(#ident)))
```

Comment:
To make our macros more robust please always use full paths:
https://doc.rust-lang.org/core/macro.stringify.html
E.g. in this case `::core::stringify!(#ident)`

Reply:
You can apply this to all occurrences of macros, functions etc.

Reply:
For trait methods we should always use `<X as Y>::z(..)` syntax to disambiguate on a semantic level.

Reply:
When referring to primitive types such as `bool`, `i32` and `str` we should always expand to types found in https://doc.rust-lang.org/core/primitive/index.html.

Reply:
> For trait methods we should always use `<X as Y>::z(..)` syntax to disambiguate on a semantic level.

Which trait method call are you referring to?

Reply:
> When referring to primitive types such as bool, i32 and str we should always expand to types found in https://doc.rust-lang.org/core/primitive/index.html.

I think this is outside the scope of this PR, it would need some considerations: e.g. how do we determine at the macro level whether a `bool` is indeed a `primitive::bool` and not a shadowing type as per the example.

Reply:
> Which trait method call are you referring to?

Was just a general rule of thumb.

> I think this is outside the scope of this PR, it would need some considerations: e.g. how do we determine at the macro level whether a bool is indeed a primitive::bool and not a shadowing type as per the example.

No, this is just needed when you literally would expand to a primitive type in a macro.

Reply:
> Was just a general rule of thumb.

Makes sense, but is there an instance here you have spotted where I call a not fully qualified trait method?

> No, this is just needed when you literally would expand to a primitive type in a macro.

I might have misunderstood the comments as being specific to this file/PR, where they are intended rather as further macro good practices.